specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/mfe.json

```diff
@@ -0,0 +1,1719 @@
+            "create",
+            "read"
+          ]
+        },
+        "protocol": {
+          "pattern": "\\w",
+          "type": "string",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        }
+      }
+    },
+    "GlusterFsSection": {
+      "required": [
+        "serverAddress",
+        "volumeName"
+      ],
+      "type": "object",
+      "properties": {
+        "serverAddress": {
+          "pattern": "\\w",
+          "type": "string",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        },
+        "volumeName": {
+          "pattern": "\\w",
+          "type": "string",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        }
+      }
+    },
+    "DatastoreContents": {
+      "required": [
+        "type"
+      ],
+      "type": "object",
+      "properties": {
+        "type": {
+          "$ref": "#/definitions/ContentsType"
+        },
+        "azureDataLake": {
+          "$ref": "#/definitions/AzureDataLakeSection"
+        },
+        "azureMySql": {
+          "$ref": "#/definitions/AzureMySqlSection"
+        },
+        "azurePostgreSql": {
+          "$ref": "#/definitions/AzurePostgreSqlSection"
+        },
+        "azureSqlDatabase": {
+          "$ref": "#/definitions/AzureSqlDatabaseSection"
+        },
+        "azureStorage": {
+          "$ref": "#/definitions/AzureStorageSection"
+        },
+        "glusterFs": {
+          "$ref": "#/definitions/GlusterFsSection"
+        }
+      }
+    },
+    "OriginType": {
+      "enum": [
+        "Synapse"
+      ],
+      "type": "string"
+    },
+    "LinkedInfo": {
+      "type": "object",
+      "properties": {
+        "linkedId": {
+          "type": "string"
+        },
+        "linkedResourceName": {
+          "type": "string"
+        },
+        "origin": {
+          "$ref": "#/definitions/OriginType"
+        }
+      }
+    },
+    "DatastoreProperties": {
+      "required": [
+        "contents"
+      ],
+      "type": "object",
+      "properties": {
+        "contents": {
+          "$ref": "#/definitions/DatastoreContents"
+        },
+        "description": {
+          "type": "string"
+        },
+        "hasBeenValidated": {
+          "type": "boolean",
+          "readOnly": true
+        },
+        "id": {
+          "format": "uuid",
+          "type": "string",
+          "readOnly": true
+        },
+        "isDefault": {
+          "type": "boolean",
+          "readOnly": true
+        },
+        "linkedInfo": {
+          "$ref": "#/definitions/LinkedInfo"
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "DatastorePropertiesResource": {
+      "description": "Azure Resource Manager resource Envelope",
+      "type": "object",
+      "properties": {
+        "properties": {
```

Comment:
properties should be marked as required unless you accept a resource with no properties